doc/contributing/releases.md

```diff
@@ -605,10 +605,14 @@ the build before moving forward. Use the following list as a baseline:
 * Run the test suite against the built binaries (optional)
 
 ```console
-./tools/test.py --shell ~/Downloads/node-v18.5.0-linux-x64/bin/node
+ # some tests are relying on the filesystem and the binary name
+ cp ~/Downloads/node-v18.5.0-linux-x64/bin/node ./out/Release/node
+./tools/test.py --shell ./out/Release/node
```

Comment:
Shouldn't we fix those tests instead? It seems to me they are using wrong assumptions, and IMO the tests should work whereever the binary happen to be.

Reply:
That's doable. I will evaluate the efforts.

Reply:
Basically, there are two failing tests using wrong assumptions. However, might be difficult to fix those 2

1) ` test/parallel/test-child-process-stdio-overlapped.js` - This one is fixed using the workflow of this PR. This test makes use of the `overlapped-check` binary which exists on `./out/Release/`.
2)  `test/parallel/test-process-config.js` - This one is relying on `config.gypi`. Regardless of the binary path, this test will keep failing because the binary doesn't contain all the variables expected by that file.

The other tests I mentioned in this PR (`test/abort/test-addon-*) are fixable by running `make build-addons`.

Therefore, I think it's ok to have two expected failures, I'll revert the changes and mention `make build-addons` and the failures.

Reply:
> 1. `test/parallel/test-child-process-stdio-overlapped.js` - This one is fixed using the workflow of this PR. This test makes use of the `overlapped-check` binary which exists on `./out/Release/`.

I've opened https://github.com/nodejs/node/pull/45015 to skip this test when the binary is not available.


> 2\. `test/parallel/test-process-config.js` - This one is relying on `config.gypi`. Regardless of the binary path, this test will keep failing because the binary doesn't contain all the variables expected by that file.

This should be fixable by removing `config.gypi` before running the test: https://github.com/nodejs/node/blob/e0ab8dd63731d8eb92053776c5160fb190d5aea5/test/parallel/test-process-config.js#L45-L46
Maybe that would be OK to document that? `config.gypi` can be restored after the facts by running `./configure`.


> I think it's ok to have two expected failures

That's certainly a valid way to think about it, however ideally we would have a way to test those binaries that doesn't involve hacks or false negatives.

Reply:
> Maybe that would be OK to document that? config.gypi can be restored after the facts by running ./configure.

+1. Code updated.